qiskit_experiments/library/characterization/t2hahn.py

```diff
@@ -133,44 +125,30 @@ def circuits(self) -> List[QuantumCircuit]:
         Returns:
             The experiment circuits.
         """
-
-        dt_unit = False
-        if self.backend:
-            dt_factor = self._backend_data.dt
-            dt_unit = dt_factor is not None
+        timing = BackendTiming(self.backend)
 
         circuits = []
         for delay_gate in np.asarray(self.experiment_options.delays, dtype=float):
-            if dt_unit:
-                delay_dt = round(delay_gate / dt_factor)
-                real_delay_in_sec = delay_dt * dt_factor
-            else:
-                real_delay_in_sec = delay_gate
-
-            total_delay = real_delay_in_sec * (self.experiment_options.num_echoes * 2)
+            num_delays = self.experiment_options.num_echoes + 1
+            # Equal delay is put before and after each echo, so each echo gets
+            # two delay gates. When there are multiple echoes, the total delay
+            # between echoes is 2 * single_delay, made up of two delay gates.
+            single_delay = delay_gate / num_delays / 2
 
             circ = QuantumCircuit(1, 1)
 
             # First X rotation in 90 degrees
             circ.rx(np.pi / 2, 0)  # Brings the qubit to the X Axis
-            for _ in range(self.experiment_options.num_echoes):
-                if dt_unit:
-                    circ.delay(delay_dt, 0, "dt")
-                    circ.rx(np.pi, 0)
-                    circ.delay(delay_dt, 0, "dt")
-                else:
-                    circ.delay(delay_gate, 0, "s")
-                    circ.rx(np.pi, 0)
-                    circ.delay(delay_gate, 0, "s")
-
-            # if number of echoes is 0 then just apply the delay gate
             if self.experiment_options.num_echoes == 0:
-                if dt_unit:
-                    total_delay = real_delay_in_sec
-                    circ.delay(delay_dt, 0, "dt")
-                else:
-                    total_delay = real_delay_in_sec
-                    circ.delay(delay_gate, 0, "s")
+                # if number of echoes is 0 then just apply the delay gate
+                circ.delay(timing.round_delay(time=delay_gate), 0, timing.delay_unit)
+                total_delay = timing.delay_time(time=delay_gate)
+            else:
+                for _ in range(self.experiment_options.num_echoes):
+                    circ.delay(timing.round_delay(time=single_delay), 0, timing.delay_unit)
```

Comment:
Probably it's better to define rounded delay value not to call round function multiple times? This is negligible performance bottleneck though. Just a nitpick of readability vs efficiency.

Reply:
Hmm, yes, probably negligible performance wise, but your comment made me think about how this experiment was working compared to others and I realized that it should be using a template circuit and assigning a parameter for the delays. When building the circuits this way, it is natural to call `round_delay` only one time.